src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -155,10 +155,334 @@ public SyntaxNode MethodDeclaration(IMethodSymbol method, IEnumerable<SyntaxNode
         }
 
         /// <summary>
-        /// Creates a method declaration.
+        /// Creates an implicit conversion declaration.
+        /// </summary>
+        public virtual SyntaxNode ImplicitConversionDeclaration(
+            IEnumerable<SyntaxNode> parameters = null,
+            SyntaxNode returnType = null,
+            Accessibility accessibility = Accessibility.NotApplicable,
+            DeclarationModifiers modifiers = default(DeclarationModifiers),
+            IEnumerable<SyntaxNode> statements = null)
+        {
+            throw new NotImplementedException();
```

Comment:
I think there's a lot more subtlety here than someone simply goofing up. 

In many circumstances, a NotSupported (commonly derived PlatformNotSupported) is thrown in a situation where it is not possible to adjust an external contract (base type or reference assembly) to add a corresponding capability API. 

Also, while strictly the right thing to do, there are costs associated with maintaining extra capability API and it is often the case that the caller has some external control over the situation (which I think is the case here, i.e. they only use particular derived types from same version in practice.)  Deferring that cost can therefore be reasonable. Throwing NotSupported now allows the capability pattern to be added later should a real scenario arise without also breaking the exception type that is thrown.